lms/djangoapps/instructor/enrollment.py

```diff
@@ -229,16 +231,25 @@ def reset_student_attempts(course_id, student, module_state_key, delete_module=F
                 except StudentModule.DoesNotExist:
                     # If a particular child doesn't have any state, no big deal, as long as the parent does.
                     pass
+        # Some blocks (openassessment) use StudentModule data as a key for internal submission data.
+        # Inform these blocks of the reset and allow them to handle their data.
+        clear_submission = getattr(block, "clear_submission", None)
+        if callable(clear_submission):
+            clear_submission(
+                user_id=user_id,
+                course_id=course_id.to_deprecated_string(),
+                item_id=module_state_key.to_deprecated_string()
+            )
+            submission_cleared = True
     except ItemNotFoundError:
         log.warning("Could not find %s in modulestore when attempting to reset attempts.", module_state_key)
 
-    # Reset the student's score in the submissions API
-    # Currently this is used only by open assessment (ORA 2)
+    # Reset the student's score in the submissions API, if the xblock.clear_submission() above has not done so already
     # We need to do this *before* retrieving the `StudentModule` model,
     # because it's possible for a score to exist even if no student module exists.
-    if delete_module:
+    if delete_module and not submission_cleared:
```

Comment:
We will have to call out in the documentation for clear_submission that implementors of the method are response for calling reset_score on the submission api (since we won't do it here).

I'm not at all fond of the complexity here, but if other xblocks are using the submission API, I guess we are stuck with it.